src/components/consent_group_list/ConsentGroupAddEdit.tsx

```diff
@@ -194,11 +191,22 @@ export const ConsentGroupAddEdit: React.FC<ConsentGroupAddEditProps> = ({
     closeAction()
   }
 
+  let headerTitle: string | undefined
+  if (readOnly) {
+    headerTitle = consentGroup?.consentGroupName
+  }
+  else if (consentGroup === undefined) {
+    headerTitle = 'New Dataset'
+  }
+  else {
+    headerTitle = `Edit ${consentGroup.consentGroupName}`
+  }
```

Comment:
I see this in other components - maybe there's a way to pull out header title calculation.